src/main/python/pybuilder/pluginloader.py

```diff
@@ -99,18 +97,40 @@ def load_plugin(self, project, name):
         raise last_problem
 
 
-def _install_external_plugin(name, logger):
+def _install_external_plugin(name, version, logger):
     if not name.startswith(PYPI_PLUGIN_PROTOCOL):
         message = "Only plugins starting with '{0}' are currently supported"
         raise MissingPluginException(name, message.format(PYPI_PLUGIN_PROTOCOL))
     plugin_name_on_pypi = name.replace(PYPI_PLUGIN_PROTOCOL, "")
     log_file = tempfile.NamedTemporaryFile(delete=False).name
-    result = execute_command(
-        'pip install {0}'.format(plugin_name_on_pypi),
-        log_file,
-        error_file_name=log_file,
-        shell=True)
+    install_cmd = ['pip', 'install']
+    if version:
+        install_cmd += ['--upgrade', '--pre', plugin_name_on_pypi + str(version)]
```

Comment:
are `upgrade` and `pre` needed when installing a specific version? I thought that was not the case.